### PR TITLE
Fix validation custom error asterisk field

### LINKED ISSUE #6245: Bug: Custom validation error for a field with an asterisk.
### CodeIgniter4 Version

dev

### What happened?

If you specify a custom error message for a field with an asterisk, the default error text will be used.
This is due to the fact that a field without an asterisk is passed to generate an error (it contains the full path of the key)

### Steps to Reproduce

```php
$data = [
    'user' => [
        ['name' => null],
        ['name' => null],
    ]
];

$validator = Services::validation();

$validator->setRules(['user.*.name' => 'required'], ['user.*.name' => ['required' => 'Required']]);

$validator->run($data);

dd($validator->getErrors());
// array (2) [
//     'user.0.name' => 'The user.*.name field is required.'
//     'user.1.name' => 'The user.*.name field is required.'
// ]
```

### Expected Output

```php
array (2) [
    'user.0.name' => 'Required'
    'user.1.name' => 'Required'
]
```

### PR DESCRIPTION
**Pull request by ping-yee**
**Description**
Fixes #6245 
Supersedes #6352 

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: user_guide_src/source/changelogs/v4.2.5.rst [hunk 1]
```diff
@@ -14,6 +14,7 @@ BREAKING
 
 - The method signature of ``BaseConnection::tableExists()`` has been changed. A second optional parameter ``$cached`` was added. This directs whether to use cache data or not. Default is ``true``, use cache data.
 - The abstract method signature of ``BaseBuilder::_listTables()`` has been changed. A second optional parameter ``$tableName`` was added. Providing a table name will generate SQL listing only that table.
+- The method signature of ``Validation.php::processRules()`` and ``Validation.php::getErrorMessage()`` have been changed. Both of these methods add new ``$originalField`` parameter.
```
Comment (kenjis) [inline user_guide_src/source/changelogs/v4.2.5.rst:17]:
```suggestion
- The method signature of ``Validation::processRules()`` and ``Validation::getErrorMessage()`` have been changed. Both of these methods add new ``$originalField`` parameter.
```

Comment (ping-yee) [reply user_guide_src/source/changelogs/v4.2.5.rst]: I have been changed!

**Comment by ping-yee** [comment]
May I `rebase` to solve resolve conflicts?

**Comment by kenjis** [comment]
> May I rebase to solve resolve conflicts?

Yes, please.

**Comment by kenjis** [comment]
Thanks, but it is not "rebase", because you added another commit. 
How did you do?

See https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/workflow.md#updating-your-branch

**Comment by ping-yee** [comment]
@kenjis I reset my git flow and re-rebase, is there the git flow show normally this time?
After I finish re-rebase, I added the test for validation.

**Review by kenjis — APPROVED** [review]

**Comment by kenjis** [comment]
Thank you. It looks good.
If all GitHub Action checks pass, I will merge this.

**Comment by kenjis** [comment]
@ping-yee Please fix coding style.
See https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md#php-style

```diff
   1) CodeIgniter4/tests/system/Validation/ValidationTest.php (trailing_comma_in_multiline, unary_operator_spaces, not_operator_with_successor_space)
      ---------- begin diff ----------
--- /home/runner/work/CodeIgniter4/CodeIgniter4/tests/system/Validation/ValidationTest.php
+++ /home/runner/work/CodeIgniter4/CodeIgniter4/tests/system/Validation/ValidationTest.php
@@ -472,7 +472,7 @@
             'foo' => [
                 ['bar' => null],
                 ['bar' => null],
-            ]
+            ],
         ];
         $this->validation->setRules(
             ['foo.*.bar' => ['label' => 'foo bar', 'rules' => 'required']],
@@ -481,7 +481,7 @@
         $this->validation->run($data);
         $this->assertSame([
             'foo.0.bar' => 'Required',
-            'foo.1.bar' => 'Required'
+            'foo.1.bar' => 'Required',
         ], $this->validation->getErrors());
     }
 

      ----------- end diff -----------
```

**Review by kenjis — CHANGES_REQUESTED** [review]
Please fix coding style.

**Comment by ping-yee** [comment]
@kenjis I have been fix my coding style.
Thank for your continuous helping, I’m very grateful.

**Comment by kenjis** [comment]
@ping-yee Thank you!

**Pull request merged**